Model/RecommendManagement.php

```diff
@@ -0,0 +1,137 @@
+{
+    /**
+     * @var null|RecommendClient
+     */
+    private $client = null;
+
+    /**
+     * @param AlgoliaHelper $algoliaHelper
+     * @param ConfigHelper $configHelper
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        private readonly AlgoliaHelper         $algoliaHelper,
+        private readonly ConfigHelper          $configHelper,
+        private readonly StoreManagerInterface $storeManager
+    ){}
+
+    /**
+     * @return RecommendClient
+     */
+    private function getClient(): RecommendClient
+    {
+        if ($this->client === null) {
+            $this->client = RecommendClient::create(
+                $this->configHelper->getApplicationID(),
+                $this->configHelper->getAPIKey()
+            );
+        }
+        return $this->client;
+    }
+
+    /**
+     * @param string $name
+     * @return string
+     * @throws AlgoliaException
+     * @throws NoSuchEntityException
+     */
+    private function getFullIndexName(string $name): string
+    {
+        $prefix = $this->configHelper->getIndexPrefix();
+        $storeCode = $this->storeManager->getStore()->getCode();
+
+        foreach ($this->algoliaHelper->listIndexes()['items'] as $index) {
+            if ($index['name'] == $prefix . $storeCode . '_' . $name) {
+                return $index['name'];
+            }
+        }
+
+        return '';
+    }
+
+    /**
+     * @param string $productId
+     * @return array
+     * @throws AlgoliaException
+     * @throws NoSuchEntityException
+     */
+    public function getBoughtTogetherRecommendation(string $productId): array
+    {
+        return $this->getRecommendations($productId, 'bought-together', 50);
+    }
+
+    /**
+     * @param string $productId
+     * @return array
+     * @throws AlgoliaException
+     * @throws NoSuchEntityException
+     */
+    public function getRelatedProductsRecommendation($productId): array
+    {
+        return $this->getRecommendations($productId, 'related-products', 50);
+    }
+
+    /**
+     * @return array
+     * @throws AlgoliaException
+     * @throws NoSuchEntityException
+     */
+    public function getTrendingItemsRecommendation(): array
+    {
+        return $this->getRecommendations('', 'trending-items', 50);
+    }
+
+    /**
+     * @param string $productId
+     * @return array
+     * @throws AlgoliaException
+     * @throws NoSuchEntityException
+     */
+    public function getLookingSimilarRecommendation($productId): array
+    {
+        return $this->getRecommendations($productId, 'bought-together', 50);
+    }
+
+    /**
+     * @param string $productId
+     * @param string $model
+     * @param float|int $threshold
+     * @return array
+     * @throws AlgoliaException
+     * @throws NoSuchEntityException
+     */
+    private function getRecommendations(string $productId, string $model, float|int $threshold = 42): array
```

Comment:
Why are we setting this default value to `42`? Is this the answer to [Life, the Universe and Everything](https://news.mit.edu/2019/answer-life-universe-and-everything-sum-three-cubes-mathematics-0910#:~:text=The%20number%2042%20is%20especially,know%20the%20answer%20to%2042.)?

Reply:
Just spitballing here but would [0](https://www.algolia.com/doc/api-reference/api-methods/get-frequently-bought-together/#method-param-threshold) perhaps be better? To increase chances of getting something back? Also think we should provide an indicator in the PHPDoc as to what this value actually does.

Reply:
42 is my favorite number :). Just kidding. I found it in this PHP example here https://www.algolia.com/doc/rest-api/recommend/#tag/recommendations/operation/getRecommendations

Or maybe put 50 in there and not send threshold on method calls

Reply:
Perhaps whoever wrote that is a Hitchhiker's Guide to the Galaxy fan. 😂 I'm in favor of whatever might make the code cleaner but perhaps the value doesn't matter - as long as the logic works consistently for the use case of verifying the model. Have we tested that having no results on a trained model still validates successfully?

Reply:
Yes, I tested. I'll add my screenshots in here.